src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2190,18 +2190,32 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                         bool reportedError = false;
                         foreach (var arm in switchExpression.SwitchArms)
-                        {
-                            var armConversion = this.Conversions.ClassifyImplicitConversionFromExpression(arm.Value, targetType, ref useSiteDiagnostics);
-                            if (!armConversion.IsImplicit || !armConversion.IsValid)
-                            {
-                                GenerateImplicitConversionError(diagnostics, arm.Value.Syntax, armConversion, arm.Value, targetType);
-                                reportedError = true;
-                            }
-                        }
+                            tryConversion(arm.Value, ref reportedError, ref useSiteDiagnostics);
+
+                        Debug.Assert(reportedError);
+                        return;
+                    }
+                case BoundKind.UnconvertedConditionalOperator:
+                    {
+                        var conditionalOperator = (BoundUnconvertedConditionalOperator)operand;
+                        HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                        bool reportedError = false;
+                        foreach (var side in new[] { conditionalOperator.Consequence, conditionalOperator.Alternative })
```

Comment:
Consider just writing these out, there's no need to allocate an array just for this.